### PR TITLE
[beat] Create beat package

### PR DESCRIPTION
**Pull request by klacabane**
### Summary

Closes https://github.com/elastic/kibana/issues/144995

Creates a Beat package that supports monitoring of standalone beats and apm-server from Agent.

### Testing

- see https://github.com/elastic/kibana/pull/146184 for e2e testing
- the package implements static and system tests. System tests are configured with two variants, a metricbeat service and an apm-server

### CODE REVIEW

**Review by crespocarlos — COMMENTED** [review]
Cool. I've quickly looked at it, but I haven't run any tests yet. First, I've got a question on the package name: `beats` or `beat`?

File: packages/beat/_dev/build/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Beat
+
+The `beat` package collects metrics and logs of Beats and APM server.
```
[the remark this answers is not in the document]
Comment (crespocarlos) [reply packages/beat/_dev/build/docs/README.md:3]: ah! got it.

**Review by miltonhultgren — APPROVED** [review]
LGTM 👍🏼 Tested and works well for both variants

File: packages/beat/manifest.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+version: 0.0.1
+description: Beat Integration
+type: integration
+release: experimental
+conditions:
+  kibana:
+    version: ^8.7.0
+owner:
+  github: elastic/infra-monitoring-ui
+format_version: 1.0.0
```
Comment (jsoriano) [inline packages/beat/manifest.yml:12]: Please use latest package spec version for a new package.
```suggestion
format_version: 2.2.0
```

File: packages/beat/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+name: beat
+title: Beat
+version: 0.0.1
+description: Beat Integration
+type: integration
+release: experimental
+conditions:
+  kibana:
+    version: ^8.7.0
```
Comment (jsoriano) [inline packages/beat/manifest.yml:9]: Try this to solve the issue with CI:
```suggestion
  kibana.version: ^8.7.0
```

**Comment by jsoriano** [comment]
Some failures now are related to the upgrade of the format version, look here for troubleshooting them: https://github.com/elastic/elastic-package/blob/main/docs/howto/update_major_package_spec.md#troubleshooting-upgrades-from-package-spec-v1-to-v2

File: packages/beat/manifest.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+name: beat
+title: Beat
+version: 0.0.1
+description: Beat Integration
+type: integration
+release: experimental
```
Comment (jsoriano) [inline packages/beat/manifest.yml:6]: Please remove also this `release` tag in the main manifest. It has been log deprecated and we will remove it definitely in package spec 2.3.0.

File: packages/beat/manifest.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+name: beat
+title: Beat
+version: 0.0.1
+description: Beat Integration
+type: integration
+conditions:
+  kibana.version: ^8.7.0
```
Comment (jsoriano) [inline packages/beat/manifest.yml:7]: You can add a condition here if you would like to keep the equivalent to `license: basic` in the format version 1.0.0.
```suggestion
  kibana.version: ^8.7.0
  elastic.subscription: basic
```

**Comment by klacabane** [comment]
System tests are complaining about the dataset format when we overwrite it in the [manifest.yml](https://github.com/elastic/integrations/blob/a1f628e94e9cfef7998dfecdacb5d3aded520b19/packages/beat/data_stream/stats/manifest.yml#L3), we may not check that field when doing the comparison

```
[0] field "data_stream.dataset" should have value "beat.state", it has "beats.stack_monitoring.state"
[1] field "event.dataset" should have value "beat.state", it has "beats.stack_monitoring.state"
```

**Comment by jsoriano** [comment]
> System tests are complaining about the dataset format when we overwrite it in the [manifest.yml](https://github.com/elastic/integrations/blob/a1f628e94e9cfef7998dfecdacb5d3aded520b19/packages/beat/data_stream/stats/manifest.yml#L3), we may not check that field when doing the comparison. edit: it should probably be picked up [here](https://github.com/elastic/elastic-package/blob/main/internal/testrunner/runners/system/runner.go#L465-L470)
> 
> ```
> [0] field "data_stream.dataset" should have value "beat.stats", it has "beats.stack_monitoring.stats"
> [1] field "event.dataset" should have value "beat.stats", it has "beats.stack_monitoring.stats"
> ```

Yeah, it looks like something is missing there. I will take a look.

**Comment by jsoriano** [comment]
@klacabane elastic-package 0.70.1 should fix the issue with dataset checks, we have just updated to this version the integrations repository, please try to update this branch.

**Pull request merged**